packages/hub/indexing/document-builder.js

```diff
@@ -0,0 +1,206 @@
+const log = require('@cardstack/logger')('cardstack/hub/indexing');
+const Model = require('../model');
+const { uniqBy } = require('lodash');
+
+module.exports = class DocumentBuilder {
+  constructor({ schema, type, id, doc, fieldMapping, getResource }) {
+    this.schema = schema;
+    this.type = type;
+    this.id = id;
+    this.doc = doc;
+    this._fieldMapping = fieldMapping ? fieldMapping : fieldName => fieldName; //TODO need a less ES specific way of approaching this
```

Comment:
I don't think we need to keep fieldMapping. If I'm remembering right, it's only used to deal with ES schema quirks.

Reply:
yup, my plan is to factor this out of here.